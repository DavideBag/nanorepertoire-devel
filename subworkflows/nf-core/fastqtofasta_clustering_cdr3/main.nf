// this subworkflow prepares the inputs from fastq files for the translation
// modules to include in this subworkflow

include {CDHIT_CDHIT   } from '../../../modules/nf-core/cdhit/cdhit/main.nf'
include {READCDHIT     } from '../../../modules/local/readcdhit/main.nf'
include {GETCDR3       } from '../../../modules/local/getcdr3/main.nf'


///Users/bagordo/Desktop/all/all_bioinformatics/nf-core-nanorepertoire/data/*_{1,2}_dummy2.fastq
// main workflow
workflow FASTQTOFASTA_CLUSTERING_CDR3 {

    // with take we define the input channels
    take:
    input      // channel: [[id], [reads_forward, reads_reverse]]
    single_end

    main:
    // define the channels that will be used to store the versions of the software

    ch_versions = Channel.empty()
    input_ch = input


    CDHIT_CDHIT(input_ch)
    ch_versions = ch_versions.mix(CDHIT_CDHIT.out.versions.first())

    READCDHIT(CDHIT_CDHIT.out.clusters, single_end = true)
    ch_versions = ch_versions.mix(READCDHIT.out.versions.first())

    GETCDR3(CDHIT_CDHIT.out.fasta, single_end = true)
    ch_versions = ch_versions.mix(GETCDR3.out.versions.first())




    // defining the output channels that the workflow will
    emit:
    // emitted channels
    cluster          = CDHIT_CDHIT.out.fasta              // channel: [[id], [ fastq_merged ]]
    clusteread       = READCDHIT.out.summary           // channel: [[id], [ fastq_renamed]]
    cdrpredicted     = GETCDR3.out.fasta
    cdrpredicted     = GETCDR3.out.tsv
   
    versions      = ch_versions                     // channel: [ versions.yml ]
}

