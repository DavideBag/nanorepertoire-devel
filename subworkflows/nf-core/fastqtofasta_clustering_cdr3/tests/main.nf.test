// nf-core subworkflows test fastqtofasta_clustering_cdr3
nextflow_workflow {

    name "Test Subworkflow FASTQTOFASTA_CLUSTERING_CDR3"
    script "../main.nf"
    workflow "FASTQTOFASTA_CLUSTERING_CDR3"
    config "./nextflow.config"
    
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastqtofasta_clustering_cdr3"
    tag "cdhit"
    tag "readcdhit"
    tag "getcdr4"


    test("translate-clustering-reading-cdr3") {

        when {
            workflow {
                """
                // input_ch
                input[0] = [
                    [id:'test'],
                    file('https://github.com/lescailab/testdata/blob/master/nanobodies/test01_aa.fasta', checkIfExists: true)
                ]
                input[1] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
