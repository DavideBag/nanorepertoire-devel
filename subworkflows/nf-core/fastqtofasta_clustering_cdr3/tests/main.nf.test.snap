{
    "translate-clustering-reading-cdr3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta:md5,1695c2cacd09c498be614659a0886b37"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.summary:md5,e79147ef0b91e56c0dbd206f4d45c88f"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_cdr3.tsv:md5,89ac4696390cab43ed2d1e5a62d4dac8"
                    ]
                ],
                "3": [
                    {
                        "id": "test"
                    }
                ],
                "4": [
                    "versions.yml:md5,2684ad5f43bf2afaf33fdbd1a001666c",
                    "versions.yml:md5,6573a2b7a84b9b1ccf5c0717ff0ba77c",
                    "versions.yml:md5,9247df97c268cf5278db5e0301fab030"
                ],
                "cdrmeta": [
                    {
                        "id": "test"
                    }
                ],
                "cdrpredicted": [
                    [
                        {
                            "id": "test"
                        },
                        "test_cdr3.tsv:md5,89ac4696390cab43ed2d1e5a62d4dac8"
                    ]
                ],
                "cluster": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta:md5,1695c2cacd09c498be614659a0886b37"
                    ]
                ],
                "clusteread": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.summary:md5,e79147ef0b91e56c0dbd206f4d45c88f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2684ad5f43bf2afaf33fdbd1a001666c",
                    "versions.yml:md5,6573a2b7a84b9b1ccf5c0717ff0ba77c",
                    "versions.yml:md5,9247df97c268cf5278db5e0301fab030"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T16:25:37.219908"
    }
}